packages/common-all/src/types/DWorkspaceV2.ts

```diff
@@ -15,8 +15,8 @@ export type DWorkspaceV2 = {
    */
   wsRoot: string;
   type: WorkspaceType;
-  config: DendronConfig;
-  vaults: DVault[];
+  config: PromiseLike<DendronConfig>;
+  vaults: PromiseLike<DVault[]>;
```

Comment:
A lot of the changes in this PR comes from this.
Since we don't have a way to apply real time changes to the config that `workspace` is holding, we have to let the config service grab it each time we try to get it.

So now the getter for config has to be awaited.

My choices here was
1. don't have config as a property of the workspace, in all places we access it (e.g. `ExtensionProvider.getDWorkspace().config`, `this._extension.getDWorkspace().config`), Do `ConfigService.instance().readConfig()` instead
2. keep it a property of workspace and make it async

I went with the latter to make minimal changes to the foundation structure.

In the future, we can go back to having objects holding a copy of the config as props once we implement a pub-sub mechanism in the config service. That way the workspace can subscribe to it and update its version of the config when a config change event is broadcasted.

Reply:
If we decide to keep it like this, yes. we don't need to do that. I was assuming _if_ we want to keep configs in these instances, we would want to do that.

I actually don't really love the idea of having the getters return `PromiseLike`s, but that's just my preference. I would say keeping it as is and not worrying about a pub-sub queue for configs is a perfectly valid choice.